### ISSUE TITLE
ERC: finalise unit names

### ISSUE THREAD

**Issue by ethers** [ethereum/EIPs#33]
The whitepaper only has wei, szabo, finney, ether:
https://github.com/ethereum/wiki/wiki/White-Paper

web3.js has shannon etc.
https://github.com/ethereum/web3.js/blob/master/lib/utils/utils.js

These other unit names are not popular and cause confusion, recent discussion in top comment:
https://www.reddit.com/r/ethereum/comments/3to11c/eip_102_serenity_rename_gas_to_mana_vbuterin/

Is an EIP needed to formalize the unit names in web3.js?  (Or remove them from web3.js?)

**Comment by alexvandesande** [comment]
My opinion is that while I support using only ether and wei for the end user (with SI extensions) and retiring the use of all other uses, I don't think removing them from web3.js is the solution. End users shouldn't be caring about variable inside the web3 object.

In numismatic, the face of a person on a coin or a banknote is called "Effigy". So here's a compromise solution: officially the only **units** become ether and wei, and we call all others **suggested effigies".  This allows the names to be kept for historical reasons and niche discussions (for example, when deciding what would be the default gas price at launch, calling them "shannon" was very useful).

So 0.000001 *ether* can be described using the *unit* **micro ether** while *shannon* is just considered a *suggested effigy*.

This is how it's currently described on the web3.js library:

```
 *   SI Short   SI Full        Effigy       Other
 * - kwei       femtoether     ada
 * - mwei       picoether      babbage
 * - gwei       nanoether      shannon      nano
 * - --         microether     szabo        micro
 * - --         milliether     finney       milli
 * - ether      --             --
 * - kether                    einstein     grand
 * - mether
 * - gether
 * - tether
```

**Comment by vbuterin** [comment]
I think we should avoid caring about unit names too much; avsa's table captures the current understanding pretty well and trying to formalize things beyond that will probably lead to more trouble than it's worth. I agree that "finney", "szabo" and "shannon" are quite valuable in particular kinds of discussions, and perhaps if some day ETH becomes ultravaluable we can transition to finney being the default unit of discussions (see Bitcoin's [in my view] general failure to get people to use "millibitcoins" or "bits"; that lesson was in fact the original reason behind coming up with finney etc in the first place) but while that day hasn't come not using finney that much seems fine.

**Comment by frozeman** [comment]
I would keep it as is as well.

**Comment by ethers** [comment]
The point is that there should be a reference for the units; having that reference be some file in web3.js is not clear (or a 3rd-party site that might not be maintained).  Here are additional discussions about the lack of clarity, and opinions that there's too many units (I have omitted arguments over the names of the units).

https://www.reddit.com/r/ethereum/comments/3j68xx/chart_of_units/

https://www.reddit.com/r/ethereum/comments/3kambp/ethereumunits_a_cli_utility_for_unit_conversion/

https://www.reddit.com/r/ethereum/comments/22eyfy/suggestion_to_simplify_coin_denominations/

https://www.reddit.com/r/ethereum/comments/3786y9/units_wikipedia_converter/

https://www.reddit.com/r/ethereum/comments/2hm4a4/eli5_why_si_units_are_not_superior_to_the/


Since the whitepaper is a wiki, should whoever wants these units back in the whitepaper just add it back?  (I agree that EIP can be bypassed.)  Initially the whitepaper had extra units, but they were removed, apparently for clarity and left the units as: "wei, szabo, finney, ether".

**Comment by alexvandesande** [comment]
I've modified [the github to remove from the main reference](https://github.com/ethereum/homestead-guide/blob/master/source/using-ethereum-the-basics/ether.rst) the mentions of some of less practical units (like the gigaether):

I believe this issue can be closed.

**Comment by romanman** [comment]
Do you want to make professional
design symbols for the units ?
ᐧ

**Comment by ethers** [comment]
thanks @alexvandesande 
@frozeman https://github.com/ethereum/web3.js/pull/240 has been updated; notably babbage and ada/lovelace switched
